stdlib/3/xmlrpc/client.pyi

```diff
@@ -0,0 +1,269 @@
+    __call_list: List[Tuple[str, Tuple[_Marshallable, ...]]]
+
+    def __init__(self, server: ServerProxy) -> None: ...
+    def __getattr__(self, item: str) -> _MultiCallMethod: ...
+    def __call__(self) -> MultiCallIterator: ...
+
+# A little white lie
+FastMarshaller: Optional[Marshaller]
+FastParser: Optional[ExpatParser]
+FastUnmarshaller: Optional[Unmarshaller]
+
+def getparser(use_datetime: bool = ..., use_builtin_types: bool = ...) -> Tuple[ExpatParser, Unmarshaller]: ...
+def dumps(params: Union[Fault, Tuple[_Marshallable, ...]], methodname: Optional[str] = ..., methodresponse: Optional[bool] = ..., encoding: Optional[str] = ..., allow_none: bool = ...) -> str: ...
+def loads(data: str, use_datetime: bool = ..., use_builtin_types: bool = ...) -> Tuple[Tuple[_Marshallable, ...], Optional[str]]: ...
+
+def gzip_encode(data: bytes) -> bytes: ...  # undocumented
+def gzip_decode(data: bytes, max_decode: int = ...) -> bytes: ...  # undocumented
+
+class GzipDecodedResponse(gzip.GzipFile):  # undocumented
+
+    io: io.BytesIO
+
+    def __init__(self, response: IO[bytes]) -> None: ...
+    def close(self) -> None: ...
+
+class _Method:  # undocumented
+
+    __send: Callable[[str, Tuple[_Marshallable, ...]], _Marshallable]
+    __name: str
+
+    def __init__(self, send: Callable[[str, Tuple[_Marshallable, ...]], _Marshallable], name: str) -> None: ...
+    def __getattr__(self, name: str) -> _Method: ...
+    def __call__(self, args: _Marshallable) -> None: ...
+
+class Transport:
+
+    user_agent: str = ...
+    accept_gzip_encoding: bool = ...
+    encode_threshold: Optional[int] = ...
+
+    _use_datetime: bool
+    _use_builtin_types: bool
+    _connection: Tuple[Optional[_HostType], Optional[http.client.HTTPConnection]]
+    _headers: List[Tuple[str, str]]
+    _extra_headers: List[Tuple[str, str]]
+
+    def __init__(self, use_datetime: bool = ..., use_builtin_types: bool = ..., *, headers: Iterable[Tuple[str, str]] = ...) -> None: ...
+    def request(self, host: _HostType, handler: str, request_body: Union[bytes, Iterable[bytes], str], verbose: bool = ...) -> Tuple[_Marshallable, ...]: ...
+    def single_request(self, host: _HostType, handler: str, request_body: Union[bytes, Iterable[bytes], str], verbose: bool = ...) -> Tuple[_Marshallable, ...]: ...
+    def getparser(self) -> Tuple[ExpatParser, Unmarshaller]: ...
+    def get_host_info(self, host: _HostType) -> Tuple[str, List[Tuple[str, str]], Dict[str, str]]: ...
+    def make_connection(self, host: _HostType) -> http.client.HTTPConnection: ...
+    def close(self) -> None: ...
+    def send_request(self, host: _HostType, handler: str, request_body: Union[bytes, Iterable[bytes], str], debug: bool) -> http.client.HTTPConnection: ...
+    def send_headers(self, connection: http.client.HTTPConnection, headers: List[Tuple[str, str]]) -> None: ...
+    def send_content(self, connection: http.client.HTTPConnection, request_body: Union[bytes, Iterable[bytes], str]) -> None: ...
```

Comment:
You're right. Its use for connection accepts Union[bytes, Iterable[bytes], str], this expects `len()`. Does Sequence work, or should I just put bytes/str?

Reply:
From the code for `send_content` I don't think a Sequence would work, because the content-length header it sends would be wrong. I feel like a str containing multibyte characters would not work either. Not familiar with this module though.

Reply:
Hm, yeah. I think only bytes is the safest type then, as that's the only one where Content-Length is definitely equal to number of bytes sent